src/django_upgrade/fixers/settings_database_posgresql.py

```diff
@@ -0,0 +1,52 @@
+"""
+Update settings.DATABASES backend path 'django.db.backends.postgresql_psycopg2'
+to the new 'django.db.backends.postgresql'.
+https://docs.djangoproject.com/en/2.0/releases/2.0/#id1
+"""
+from __future__ import annotations
+
+import ast
+from typing import Iterable
+
+from tokenize_rt import Offset, Token
+
+from django_upgrade.ast import ast_start_offset
+from django_upgrade.data import Fixer, State, TokenFunc
+from django_upgrade.tokens import replace, str_repr_matching
+
+fixer = Fixer(
+    __name__,
+    min_version=(1, 9),
+)
+
+
+@fixer.register(ast.Dict)
+def visit_Dict(
+    state: State,
+    node: ast.Dict,
+    parents: list[ast.AST],
+) -> Iterable[tuple[Offset, TokenFunc]]:
+    if (
+        state.looks_like_settings_file()
+        and isinstance(parents[-1], ast.Dict)
+        and isinstance((db_setting := parents[-2]), ast.Assign)
+        and isinstance(db_setting.targets[0], ast.Name)
+        and db_setting.targets[0].id == "DATABASES"
+        and len(db_setting.targets) == 1
```

Comment:
length checks should be done before accessing list indexes, to avoid crashes where the list isn't the required length. I don't think it's possible to have a crash here, though:

* `parents[-1]` will always exist, and if it's a `Dict`, `parents[-2]` will always exist, at least as the `ast.Module`
* I don't think it's possible for an `Assign` to have 0 targets (would be weird)

But still worth putting the length checks first to embed the right pattern, plus they should be cheaper than the `isinstance` checks.

Reply:
> I don't think it's possible for an Assign to have 0 targets (would be weird)

That's the assumption I made and the reason why I delayed the `len` check but yeah I'm not sure it's faster anyway and it's less clear so let's keep your version.

> parents[-1] will always exist, and if it's a Dict, parents[-2] will always exist, at least as the ast.Module

yeah exactly, but I agree it might be confusing. Better explicit than implicit and the performance cost wont matter anyway